### ISSUE TITLE
Better way to update LR etc in optim.Optimizer

### ISSUE THREAD

**Issue by chsasank** [pytorch/pytorch#1244]
Currently to change LR of `optimizer`, we have to do this:

```python
for param_group in optimizer.param_groups:
        param_group['lr'] = lr
```

See [here](https://discuss.pytorch.org/t/adaptive-learning-rate/320) for more discussion

Although this works, this looks as if the user is messing with the internals of `optimizer`. Therefore I propose to add a member function to `optim.Optimizer` something like this:

```python
def update_params(new_params):
    if isinstance(new_params, dict):
        for param_group in self.param_groups:
            param_group.update(new_params)
    elif isinstance(new_params, list):
        assert len(new_params) == len(self.param_groups)
        for param_group, new_pram in zip(self.param_groups, new_params):
            param_group.update(new_pram)
    else:
        raise TypeError
```

Sasank.

**Comment by chsasank** [comment]
I can create a PR if the above structure is fine.

**Comment by chsasank** [comment]
Fixed by #1370
